projects/spectator/test/with-routing/my-page.component.spec.ts

```diff
@@ -165,10 +166,15 @@ describe('MyPageComponent', () => {
       await spectator.fixture.whenStable();
       expect(spectator.inject(Location).path()).toBe('/');
 
-      await spectator.router.navigate(['/foo']);
+      const ngZone = TestBed.get(NgZone);
```

Comment:
Why do you need to add this code?

Reply:
When I added the zone in the create-factory.ts as you suggested it resolved all the warnings in the spec file except the navigation in this test case. This change fixes that warning. I can revert it to just the change on the create-factory.ts if you think this one is not necessary.

Reply:
But why it didn't resolved this case?

Reply:
I think it could be related to what is being discussed here https://github.com/angular/angular/issues/25837

Reply:
I think it could be related to what is being discussed here https://github.com/angular/angular/issues/25837